python_modules/dagster/dagster/_core/asset_graph_view/bfs.py

```diff
@@ -152,9 +152,9 @@ def dequeue(self) -> AssetGraphSubset:
         # (i.e. asset key).
 
         # the minimum item in a heap is always at index 0
-        min_item = self._heap[0]
+        min_item = heappop(self._heap)
+        result = min_item.asset_graph_subset
```

Comment:
this seems logically different right? because heappop actually takes it out of the heap whereas before we were just inspecting it?

Reply:
the intention is for it to just short circuit the first iteration - it is logically different but i would not expect a behavior difference in practice unless i am missing something?

Reply:
gotcha i understand now